lib/rucio/daemons/storage/consistency/actions.py

```diff
@@ -0,0 +1,805 @@
+        if once:
+            break
+
+    die(executable=executable, hostname=hostname, pid=pid, thread=current_thread)
+
+
+def stop(signum=None, frame=None):
+    """
+    Graceful exit.
+    """
+    graceful_stop.set()
+
+
+def run(once=False, rses=None, sleep_time=60, default_dark_min_age=28, default_dark_threshold_percent=1.0,
+  default_miss_threshold_percent=1.0, force_proceed=False, default_scanner_files_path="/var/cache/consistency-dump",
+  threads=1):
+    """
+    Starts up the Consistency-Actions.
+    """
+
+# Setup logging
+    setup_logging()
+
+    prefix = 'storage-consistency-actions (run())'
+    logger = formatted_logger(logging.log, prefix + '%s')
+
+    logger(logging.INFO, '\n Now inside run(once...)\n')
+
+    dark_min_age = default_dark_min_age
+    dark_threshold_percent = default_dark_threshold_percent
+    miss_threshold_percent = default_miss_threshold_percent
+    scanner_files_path = default_scanner_files_path
+    # We expect these to be sourced from the RSE config in the future.
+
+    if rses == []:
+        logger(logging.INFO, '\n NO RSEs passed. Will loop over all writable RSEs.')
+
+        rses = [rse['rse'] for rse in list_rses({'availability_write': True})]
+
+# Could limit it only to Tier-2s:
+#        rses = [rse['rse'] for rse in list_rses({'tier': 2, 'availability_write': True})]
+
+    logging.info('\n RSEs: %s' % rses)
+    logger(logging.INFO, '\n RSEs: %s  \n run once: %r \n Sleep time: %d  \n Dark min age (days): %d\
+     \n Dark files threshold %%: %f  \n Missing files threshold %%: %f  \n Force proceed: %r\
+     \n Scanner files path: %s ' % (rses, once, sleep_time, dark_min_age, dark_threshold_percent,
+     miss_threshold_percent, force_proceed, scanner_files_path))
+
+    executable = 'storage-consistency-actions'
+    hostname = socket.gethostname()
+    sanity_check(executable=executable, hostname=hostname)
+
+# It was decided that for the time being this daemon is best executed in a single thread
+# If this decicion is reversed in the future, the following line should be removed.
+    threads = 1
+
+    if once:
+        actions_loop(once, rses, sleep_time, dark_min_age, dark_threshold_percent,
+          miss_threshold_percent, force_proceed, scanner_files_path)
+    else:
+        logging.info('Consistency Actions starting %s threads' % str(threads))
+        threads = [threading.Thread(target=actions_loop, kwargs={'once': once, 'rses': rses, 'sleep_time': sleep_time,
+         'dark_min_age': dark_min_age, 'dark_threshold_percent': dark_threshold_percent,
+         'miss_threshold_percent': miss_threshold_percent, 'force_proceed': force_proceed,
+         'scanner_files_path': scanner_files_path}) for i in range(0, threads)]
+        logger(logging.INFO, '\n Threads: %d' % len(threads))
+        [t.start() for t in threads]
+        # Interruptible joins require a timeout.
+        while threads[0].is_alive():
+            [t.join(timeout=3.14) for t in threads]
```

Comment:
Is the timeout too much? 3 Seconds seems kinda long. (name of your sextape lol)

Reply:
Same as above.